src/views/bootablevolumes/actions/components/EditBootableVolumesModal.tsx

```diff
@@ -17,73 +19,99 @@ import TabModal from '@kubevirt-utils/components/TabModal/TabModal';
   preferences: V1alpha2VirtualMachineClusterPreference[];
-  instanceTypesNames: string[];
 };
 
 const EditBootableVolumesModal: FC<EditBootableVolumesModalProps> = ({
   dataSource,
   isOpen,
   onClose,
   preferences,
-  instanceTypesNames,
 }) => {
   const { t } = useKubevirtTranslation();
+
   const preferencesNames = useMemo(
     () => Object.keys(convertResourceArrayToMap(preferences)).sort((a, b) => a.localeCompare(b)),
     [preferences],
   );
 
-  const instanceTypesToSizesMap = useMemo(
-    () => getInstanceTypesToSizesMap(instanceTypesNames),
-    [instanceTypesNames],
-  );
-
   const initialParams = useMemo(() => {
     const instanceTypeLabel =
       dataSource?.metadata?.labels?.[DEFAULT_INSTANCETYPE_LABEL]?.split('.');
 
     return {
       preference: dataSource?.metadata?.labels?.[DEFAULT_PREFERENCE_LABEL],
-      instanceType: instanceTypeLabel?.[0],
+      instanceType: () => {
+        for (const category of Object.entries(categoryDetailsMap)) {
+          if (category[1].prefix === instanceTypeLabel?.[0]) {
+            return category;
+          }
+        }
+      },
       size: instanceTypeLabel?.[1],
       description: dataSource?.metadata?.annotations?.[ANNOTATIONS.description],
     };
   }, [dataSource]);
 
   const [preference, setPreference] = useState<string>(initialParams.preference);
-  const [instanceType, setInstanceType] = useState<string>(initialParams.instanceType);
+  const [isInstanceTypeOpen, setIsInstanceTypeOpen] = useState(false);
+  const [instanceType, setInstanceType] = useState<string>(initialParams.instanceType()?.[0]);
+  const [isSizeOpen, setIsSizeOpen] = useState(false);
   const [size, setSize] = useState<string>(initialParams.size);
   const [description, setDescription] = useState<string>(initialParams.description);
 
-  const sizeOptions = useMemo(
-    () => instanceTypesToSizesMap[instanceType] || [],
-    [instanceType, instanceTypesToSizesMap],
+  // update options for 'Size' dropdown according to chosen 'instanceType' in 'Default InstanceType' dropdown
+  const { instanceTypes }: CategoryDetails = useMemo(
+    () => (instanceType ? categoryDetailsMap[instanceType] : {}),
+    [instanceType],
   );
 
-  const onInstanceTypeChange = useCallback(
-    (instanceTypeName: string) => {
-      setInstanceType(instanceTypeName);
-      setSize(instanceTypesToSizesMap[instanceTypeName][0]);
+  const onInstanceTypeSelect = useCallback(
+    (_event: ChangeEvent<HTMLSelectElement>, newInstanceType: string) => {
+      setInstanceType(newInstanceType);
+      setIsInstanceTypeOpen(false);
+
+      const newCategoryObject = categoryDetailsMap[newInstanceType];
+      const newCategorySize = newCategoryObject.instanceTypes[0].label;
+      setSize(newCategorySize);
     },
-    [instanceTypesToSizesMap],
+    [],
   );
 
-  const onChangeVolumeParams = useCallback(() => {
+  const onSizeSelect = useCallback((_event: ChangeEvent<HTMLSelectElement>, newSize: string) => {
```

Comment:
There is no benefit here for useCallback as there are no dependencies nor hard calculations, but useCallback does make a comparison each time which adds complex

Reply:
Originally, there were some dependencies. Once I found I don't need them, maybe I just forgot to remove useCallback after that.